gadopt/time_stepper.py

```diff
@@ -117,120 +124,209 @@ def advance(
+            update_forcings(float(self.t))
 
-    def solve_tendency(self, i_stage, update_forcings, t) -> None:
-        """Evaluates the tendency of i-th stage"""
-        if self._nontrivial:
-            if update_forcings is not None and t is not None:
-                update_forcings(t + self.c[i_stage] * self.dt)
-            elif update_forcings is not None:
-                update_forcings()
+        # Update time if provided
+        if t is not None:
+            self.t.assign(float(t))
 
-            self.solver[i_stage].solve()
+        # Advance using Irksome
+        self.stepper.advance()
 
-    def get_final_solution(self):
-        self.solution.assign(self.solution_old)
-        if self._nontrivial:
-            self.solution += self.final_sol_expr
-        self.solution_old.assign(self.solution)
+        # Update time for next step
+        self.t.assign(float(self.t) + float(self.dt_mesh_const))
 
-    def solve_stage(self, i_stage, update_forcings, t) -> None:
-        self.update_solution(i_stage)
-        self.solve_tendency(i_stage, update_forcings, t)
+    def set_dt(self, new_dt: float):
+        """Update the time step."""
+        self.dt = float(new_dt)
+        self.dt_reference.assign(float(new_dt))
+        # Note: dt_mesh_const will be synced on next advance() call
 
 
-class DIRKGeneric(RungeKuttaTimeIntegrator):
-    """Generic implementation of Diagonally Implicit Runge Kutta schemes.
+def gadopt_to_irksome_tableau(scheme_class):
+    """Convert a G-ADOPT scheme class to an Irksome Butcher tableau.
 
-    All derived classes must define the Butcher tableau coefficients :attr:`a`,
-    :attr:`b`, :attr:`c`.
+    Args:
+        scheme_class: A G-ADOPT AbstractRKScheme class
 
-    Arguments:
-      equation: G-ADOPT equation to solve
-      solution: Firedrake function reperesenting the equation's solution
-      dt: Integration time step
-      solution_old: Firedrake function representing the equation's solution
-                      at the previous timestep
-      solver_parameters: Dictionary of solver parameters provided to PETSc
-      strong_bcs: List of Firedrake Dirichlet boundary conditions
+    Returns:
+        tuple: (ButcherTableau instance, stage_type string)
+    """
+    # Create a temporary instance to access the tableau
+    temp_scheme = scheme_class()
+
+    # Define scheme mappings
+    scheme_mappings = {
+        # Special cases with direct Irksome equivalents
+        BackwardEulerAbstract: lambda: (IrksomeBackwardEuler(), "dirk"),
+        ImplicitMidpointAbstract: lambda: (GaussLegendre(1), "dirk"),
+
+        # Explicit schemes
+        ForwardEulerAbstract: lambda: (create_custom_tableau(temp_scheme.a, temp_scheme.b, temp_scheme.c), "explicit"),
+        SSPRK33Abstract: lambda: (create_custom_tableau(temp_scheme.a, temp_scheme.b, temp_scheme.c), "explicit"),
+        ERKMidpointAbstract: lambda: (create_custom_tableau(temp_scheme.a, temp_scheme.b, temp_scheme.c), "explicit"),
```

Comment:
Are we missing `eSSPRKs3p3Abstract` and `eSSPRKs10p3Abstract` here for level-set purposes?